doc/introduction/measurements.rst

```diff
@@ -354,6 +354,53 @@ The transform can be applied to a QNode as follows:
     If the ``defer_measurements`` transform is used in analytic mode, ``backprop`` is also a viable
     option.
 
+.. _tree_traversal:
+
+The tree-traversal algorithm
+****************************
+
+Dynamic circuit execution is akin to traversing a binary tree where each MCM
+corresponds to a node and groups of gates between the MCMs correspond to edges.
+The :func:`~.pennylane.dynamic_one_shot` approach picks a branch of the tree randomly
+and simulates it from beginning to end.
+This is wasteful in many cases; the same branch is simulated many times
+when there are more shots than branches for example.
+The tree-traversal algorithm does away with such redundancy while retaining the
+exponential gains in memory of the one-shot approach compared with the deferred
+measurement principle, among other advantages.
+
+Briefly, it proceeds from the top of the tree and begins splitting the circuit
+into two: a circuit running to the next MCM replacing the MCM by a sample
+measurement, and a circuit consisting of the rest of the circuit. A key point is that
+all samples are collected at once, removing the need to run the circuit several times.
+The samples are converted to counts and a copy of the state vector is made at each node.
+We loop over the possible branches, collapsing the state vector and pursuing the calculation
+until the next MCM. Since many counts come out to be zero in practice, it is possible
+to ignore entire sub-trees, reducing the computational burden.
+We eventually reach the bottom of the tree having cached at most :math`n_{MCM}+1`
+state vector copies, which is an exponential improvement compared with
+:func:`~. pennylane.defer_measurements` requiring :math`O(2^{n_{MCM}})` memory.
+We then obtain terminal measurements, propagating and combining at each node up the
+tree. This algorithm is thus of the depth-first family of tree-traversal algorithms.
+
+To summarize, this algorithm gives us the best of both worlds. In the limit of few
+shots and/or many MCMs, it is as fast as the naive shot-by-shot implementation
+because few sub-trees are explored. In the limit of many shots and/or few MCMs, it is
+equal to or faster than the deferred measurement algorithm (albeit with more
+overheads in practice) because each tree edge is visited at most once, all while
+reducing the memory requirements exponentially.
+
+The tree-traversal algorithm is not a transform. Its usage is therefore specified
+by passing an ``mcm_method`` option to a QNode (see section
+:ref:`MCMConfig <mcm_config>`). For example,
```

Comment:
```suggestion
shots and/or many mid-circuit measurements, it is as fast as the naive shot-by-shot implementation
because few sub-trees are explored. In the limit of many shots and/or few mid-circuit measurements, it is
equal to or faster than the deferred measurement algorithm (albeit with more
overheads in practice) because each tree edge is visited at most once, all while
reducing the memory requirements exponentially.

The tree-traversal algorithm is not a transform. Its usage is therefore specified
by passing an ``mcm_method`` option to a QNode (see section
:ref:`"Configuring mid-circuit measurements" <mcm_config>`). For example,
```